build.sbt

```diff
@@ -61,6 +61,49 @@ lazy val netcdf = (project in file("netcdf"))
 		credentials += Credentials(Path.userHome / ".ivy2" / ".credentials")
 	)
 
+
+val frontend = inputKey[Unit]("Builds the frontend")
+frontend := Def.inputTaskDyn {
+	import scala.sys.process.Process
+	import java.io.File
+	val log = streams.value.log
+
+	val args: Seq[String] = sbt.Def.spaceDelimited().parsed
+
+	 Def.taskDyn {
+		args.toList match {
+			case "build" :: app :: Nil =>
+				log.info(s"Build $app")
+				val projectDirectory = new File(s"src/main/js/$app/")
+				val exitCode = (Process("npm install", projectDirectory) #&& Process("npm run build", projectDirectory)).!
+				if (exitCode == 0) {
+					log.info("Finished front-end build for " + app)
+					None
+				} else {
+					log.error(s"Front-end build for $app failed")
+					Some(s"Front end building for $app returned non-zero exit code $exitCode")
+				}
+			case _ =>
+				log.info("Nothing")
```

Comment:
log.info("Usage: frontend build `<app>`, where app is one of: " + jsApps.mkString(", "))